p2p/proto/state.proto

```diff
@@ -0,0 +1,39 @@
+syntax = "proto3";
+import "p2p/proto/common.proto";
+
+
+// optimized for flat storage, not through a trie (not sharing key prefixes)
+message ContractStoredValue {
+    Felt252 key = 1;
+    Felt252 value = 2;
+}
+
+message StateDiff
+{
+    // a bit more efficient than the state sync separation
+    message ContractDiff {
+        Address address = 1;
+        Felt252 nonce = 2;
+        Felt252 class_hash = 3; // can change for replace_class or new contract
+        repeated ContractStoredValue values = 4;
+    }
+
+    uint32 tree_id = 1;
+    repeated ContractDiff contract_diffs = 2;
+}
+
+// is it better to separate the definition from the hashes? (will need to repeate the hashes
+// for the definitions stream)
+// or, make the definitions optional? maybe it is enough to know only that a class exists, not its definition
+// which may be fetched lazily later.
+message Class {
+    Hash compiled_hash = 2; // TBD: add also/instead CASM definition (instead of every node compiling)?
+    bytes definition = 3; // compressed? size limit or split to chunks.
```

Comment:
Definitions get pretty big and imo should be compressed. Nevertheless `chunk_id` may still be necessary; I guess there's no hard limit on sierra bytecode size so even compressed it could exceed 1MiB? @ittayd

Reply:
Yeah, even if there is, best not to put it in the spec. So this means Classes can have PartialClass ? About compression, I agree the spec should be to compress. Do you  have suggestion for the compression algorithm (balancing speed & compression)?

Reply:
PartialClass could be a good idea to future proof Classes

As for compression in pathfinder we use [zstd](http://facebook.github.io/zstd/) particularly because of its speed

Reply:
> Do you have suggestion for the compression algorithm (balancing speed & compression)?
ethereum uses snappy, but zstd looks like a better alternative

Reply:
Well, we need a consensus... I don't have any preference other that it'll be implemented across as many languages as possible. 

About class chunking, how about each Class object will have a chunk_count and total_chunks optional fields? (if not specified, then assumed total_chunks is 1)